### DISCUSSION TITLE
Line links in comments

### DISCUSSION THREAD

**Discussion by InkONat** [godotengine/godot-proposals#8697] [Scripting]
I think it would be nice if you could link to a line in the script with a comment:

```
var foo : int = 2

func _ready():
        print(foo) # Prints foo, declared at Line 1
```
Line 1 would be highlighted and you could click it for the script editor to go to Line 1

This QOL change would mostly benefit scripts with several lines, and would make it easier to know what a variable does in code (by linking the lines where it is used)

I don't know if "Line X" is the best way to link a line, so I'd like a discussion on a better way of how the link would be created

**Comment by Calinou** [comment]
The issue is that line numbers can change quickly as you refactor code, making your link no longer relevant.

A better way to handle this would be to create internal links to properties or methods in the class.

**Comment by InkONat** [comment]
I agree tbh, I'll open a discussion post on that.

**Comment by InkONat** [comment]
https://github.com/godotengine/godot-proposals/discussions/8701

how do I close this discussion

**Reply by Calinou** [reply]
> how do I close this discussion

Use the **Close discussion** button in the bottom-right corner below the comment box at the bottom of the page.
